CONTRIBUTION.md

```diff
@@ -1,42 +1,52 @@
 In General a PR should explain what it does, how it does this and how it was tested. It should also follow the [Code Guidelines](#code-guidelines).
 * If you are not yet finished implementing, prepend it with \[WIP]
 * A description must be provided before the PR is merged, to provide record of the Performed changes
+
 ### Bug Fix PR's
 Your PR should include the following at least the following Information:
 * A reference to the corresponding issue or a full description of the issue being fixed
 * An explanation how this PR solves/attempts to solve the given issue
 * If testing was performed, what tests and how extensive they have been
+
 ### Feature Request PR's
 Your PR should include the following at least the following Information:
 * A reference to the corresponding issue or a full description of the feature being implemented
 * An explanation how this PR attempts to implement the given feature
 * If testing was performed, what tests and how extensive they have been
+
 ## Code Guidelines
 In General your code should be well-structured and easy to read. 
 We know that as of 1.12 this is not the case for great parts of the mod, but we are trying to improve upon the code-base.
-Also your PR should not contain any formatting only changes to Files, as this simply makes reviewing them unnecessarily complex.
-Try to avoid unnecessary commits in order not to pollute the commit history.
+Unless your Pr is directly related to code cleaning then your PR should not contain any formatting changes to unchanged code blocks as this simply makes reviewing them unnecessarily complex.
+
 ### Formatting
 * Use 4 space indent instead of Tabs
 * operators should be surrounded by spaces
-* a `,` comma to separate types should be followed by exactly one Space
+  * For example `if(statement == true)`
+* a `,` comma to separate types should be followed by exactly one space
 * type casts should be followed by exactly one space
 * `{` should not be prefixed with returns
-* Methods should have exactly one free line between them (/their docs)
+* Methods should have exactly one free line between them (or their doc blocks)
 * Avoid empty lines at the start of class, interface or enum definitions
 * Avoid unnecessary lines in empty Methods
     * `public void emptyImpl() {}`
+    * Also try and avoid introducing code that may require empty methods
 * Method/Class annotations should gain their own lines in front of the definition
 * Parameter Annotations should be surrounded by exactly one space
+* Comments shoud always start with a single space `// comment`
+
 ### Code Style
 In general apply common sense to the Code you write. "Effective Java" is also always a good start.
 * Avoid unnecessary else clauses
+* Avoid large case statements as it's likely not needed. If you find yourself requiring a large case statement block then try to think about how you might be able to refactor your code in a way that avoids the requirement. Small case statements are fine.
```

Comment:
```suggestion
* Avoid large case statements as it's likely not needed. If you find yourself requiring a large case statement block then try to think about how you might be able to refactor your code in a way that avoids the requirement. Small case statements are fine.
    * In most cases this can be avoided by giving the object that's being tested on an abstract method to be overridden. For example, give item and block objects a method called `getGuiElement(EntityPlayer, int, int ,int)` and call it in `GuiProxy` instead of using a case statement.
    * The other approach would be using a `Map<MyObject,Comsumer<MyObject>>` and setup all different cases as lambdas where the parameter is the object being tested.
```
Just some personal opinion.